rosbag2_transport/src/rosbag2_transport/rosbag2_transport_python.cpp

```diff
@@ -128,7 +163,23 @@ rosbag2_transport_play(PyObject * Py_UNUSED(self), PyObject * args, PyObject * k
   play_options.node_prefix = std::string(node_prefix);
   play_options.read_ahead_queue_size = read_ahead_queue_size;
 
-  rosbag2_transport::Rosbag2Transport transport;
+  rosbag2_storage::MetadataIo metadata_io{};
+  rosbag2_storage::BagMetadata metadata{};
```

Comment:
`metadata_io` should be moved closer to where its used at L175.
`metadata` should be declared and assigned on L176